src/main/java/com/adam/springproject/services/UserService.java

```diff
@@ -1,25 +1,26 @@
 package com.adam.springproject.services;
 
-import com.adam.springproject.dtoas.LogInRequestDto;
-import com.adam.springproject.dtoas.LoginResponseDto;
-import com.adam.springproject.dtoas.SignUpRequestDto;
-import com.adam.springproject.dtoas.SignUpResponseDto;
+import com.adam.springproject.dtoas.*;
 import com.adam.springproject.repositories.UserDao;
 import com.adam.springproject.repositories.entites.User;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
+import javax.validation.ConstraintViolationException;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import java.util.regex.Pattern;
 
 @Service
 public class UserService {
-    @Autowired
-    private UserDao userDao;
+    private final UserDao userDao;
+    private Optional<User> user;
```

Comment:
Remove this field. Don't save any dynamic states in the service class.
Your service class doesn't only server one user. It serves a lot of users.